src-electron/rest/user-data.js

```diff
@@ -111,13 +111,20 @@ function httpPostCluster(db) {
         flag
       )
 
-      if (insertDefault) {
+      // Insert cluster's defaults when insertDefault variable is not undefined/null or when the flag is true 
+      if (insertDefault || flag) {
         await queryConfig.insertClusterDefaults(db, endpointTypeId, packageId, {
           clusterRef: id,
           side: side,
         })
       }
 
+      // Delete the disabled cluster's attributes if the flag is false which means the user disabled the cluster
+      if(!flag){
+        let { endpointTypeClusterId } = await queryConfig.selectEnabledClusterState(db, endpointTypeId, id, side, 0);
+        await queryAttribute.deleteEndpointTypeAttributes(db, endpointTypeClusterId, endpointTypeId);
```

Comment:
Do not do delete attributes from the endpoint_type_attribute table. If you do this then when the user enables the same cluster then they loose their old selections.
What you want to do instead is check if the cluster is enabled or not and then count the attributes.
Avoid deleting anything from the database.

You can use the "enabled" attribute of the endpoint_type_attribute table to get what you need here.